src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheCrusader/boss_northrend_beasts.cpp

```diff
@@ -500,10 +502,49 @@ struct npc_snobold_vassal : public ScriptedAI
         }
     }
 
+    Player* SelectNewVictim()
+    {
+        const Map::PlayerList& players = me->GetMap()->GetPlayers();
+        std::vector<Player*> availablePlayers;
+        for (auto i = players.begin(); i != players.end(); ++i)
+        {
+            if (!i->GetSource()->IsAlive())
+            {
+                continue;
+            }
+
+            availablePlayers.push_back(i->GetSource());
+        }
+
+        if (availablePlayers.size() == 0)
+        {
+            return nullptr;
+        }
+
+        srand(me->GetMap()->GetInstanceId());
+        int playerIndex = rand() % availablePlayers.size();
+        return availablePlayers[playerIndex];
+    }
+
     void UpdateAI(uint32 diff) override
     {
+        if (!me->GetVictim() || !me->GetVictim()->IsAlive())
+        {
+            auto vehicle = me->GetVehicleBase();
```

Comment:
`Unit* vehicle = me->GetVehicleBase();`

only use auto for complex types for e.g. something along `std::map<x,y>::const_iterator`